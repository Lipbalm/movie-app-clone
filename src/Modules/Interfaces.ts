export interface ITopRatedMovies<T> {
  page: number;
  total_results: number;
  total_pages: number;
  results: T;
}

export interface ITopRatedResults {
  popularity: number;
  vote_count: number;
  video: boolean;
  poster_path: string;
  id: number;
  adult: false;
  backdrop_path: string;
  original_language: string;
  original_title: string;
  genre_ids: number[];
  vote_average: number;
  overview: string;
  release_date: string;
  title: string;
}

export interface IGenreIds {
  id: number;
  name: string;
}
