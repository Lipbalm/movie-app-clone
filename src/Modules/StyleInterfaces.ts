export interface IStyleProperty {
  [key: string]: string;
}
export interface ICSSPesudo {
  [key: string]: IStyleProperty;
}
export interface IStyleFrame {
  basis: IStyleProperty;
  pesudo?: ICSSPesudo;
}
export interface IStyle {
  styleObj?: IStyleFrame;
}
