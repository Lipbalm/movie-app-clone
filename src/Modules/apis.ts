import axios, { AxiosInstance } from "axios";
import {
  ITopRatedMovies,
  ITopRatedResults,
  IGenreIds,
  IMapGenreIds
} from "./Interfaces";

const apiKey = "f14b5a458d4b24203b40ecec534dfd0d";
const baseMoviesInstance: AxiosInstance = axios.create({
  baseURL: `https://api.themoviedb.org/3/movie`
});

const imageBaseURL: string = "https://image.tmdb.org/t/p/w500/";

let genreIds: IMapGenreIds[];

const getTopRatedMovie = (language: string, page: number) =>
  baseMoviesInstance
    .get<ITopRatedMovies<ITopRatedResults[]>>(
      `/top_rated?api_key=${apiKey}&language=${language}&page=${page}`
    )
    .then(response => response.data);

const getGenreIds = (language: string) =>
  axios
    .get<IGenreIds[]>(
      `https://api.themoviedb.org/3/genre/movie/list?api_key=${apiKey}&language=${language}`
    )
    .then(response => {
      response.data.forEach(v =>
        v.name === undefined ? (genreIds[v.id] = "") : (genreIds[v.id] = v.name)
      );
      return true;
    })
    .catch(e => e);

export { getTopRatedMovie, genreIds, getGenreIds, imageBaseURL };
