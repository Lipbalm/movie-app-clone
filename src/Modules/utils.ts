function isMatch(input: string, pattern: string): boolean {
  let regex = new RegExp(pattern, "g");
  return regex.test(input);
}
export { isMatch };
