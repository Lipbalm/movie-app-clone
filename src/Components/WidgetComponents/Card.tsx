import React, { FC, useState } from "react";
import { ITopRatedResults } from "../../Modules/Interfaces";
import { RectCard, ThumnailWrapper, ThumnailImg } from "../../Style";
import { baseImageURL } from "../../Modules/apis";
import CardLabel from "./CardLabel";

interface IProps {
  movie: ITopRatedResults;
  index: number;
}

const Card: FC<IProps> = ({ movie, index }) => {
  const {
    id,
    genre_ids,
    adult,
    original_title,
    title,
    poster_path,
    vote_average
  } = movie;

  const [value, setValue] = useState<string>("");

  const handleClick = () => {
    setValue("Clicked");
  };

  return (
    <RectCard
      isMargin={index > 0 ? true : false}
      data-testid="margin-test"
      onClick={handleClick}
    >
      {value}
      <ThumnailWrapper>
        <ThumnailImg src={`${baseImageURL}${poster_path}`} alt="no image" />
        <CardLabel genre_id={genre_ids[0]} name={title} />
      </ThumnailWrapper>
    </RectCard>
  );
};

export default Card;
