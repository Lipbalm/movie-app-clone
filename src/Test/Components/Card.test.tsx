import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Card from "../../Components/WidgetComponents/Card";
import { tempCard } from "../TempData/Card";
import { baseImageURL } from "../../Modules/apis";

const setup = () => {
  const utils = render(<Card card_info={tempCard} index={0} />);
  return { ...utils };
};

describe("Card Component", () => {
  it("Card SnapShot", () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it("Card 썸네일 못 불러올 경우", () => {
    const { getByAltText } = setup();
    const alt = getByAltText("no image");
    expect(alt).toBeTruthy();
  });

  it("Card 이미지 렌더 테스트", () => {
    const { getByAltText } = setup();
    const alt = getByAltText("no image");
    expect(alt).toHaveAttribute(
      "src",
      `${baseImageURL}${tempCard.poster_path}`
    );
  });

  it("Card Margin 확인", () => {
    const utils = setup();
    const { getByTestId } = utils;
    const RectCard = getByTestId("card");

    expect(RectCard).toHaveStyle(`
      margin-left: 0;
    `);
  });

  it("Card Click", () => {
    const { getByTestId } = setup();
    const RectCard = getByTestId("card");

    fireEvent.click(RectCard);

    expect(RectCard).toHaveTextContent(
      `${tempCard.genre_ids.join("/")}${tempCard.title}`
    );
  });
});
