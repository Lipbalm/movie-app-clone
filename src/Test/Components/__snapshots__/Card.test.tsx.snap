// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component Card SnapShot 1`] = `
<div>
  <div
    class="sc-jTzLTM gEiCHi"
    data-testid="margin-test"
  >
    
    <div
      class="sc-jzJRlG lhwmZp"
    >
      <img
        alt="no image"
        class="sc-cSHVUG dvKLDq"
        src="https://image.tmdb.org/t/p/w500//2CAL2433ZeIihfX1Hb2139CX0pW.jpg"
      />
    </div>
  </div>
</div>
`;
